source/particle/interpolator/quadratic_least_squares.cc

```diff
@@ -155,39 +250,398 @@ namespace aspect
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[2] = 0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[0] =  0.5;
+                        point[1] = -0.5;
+                        point[2] = -0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[2] = 0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[1] = 0.5;
+                        point[2] = -0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+                        point[2] = 0.5;
+                        value_at_corner = evaluate(c[property_index], point);
+                        interpolation_min = std::min(interpolation_min, value_at_corner);
+                        interpolation_max = std::max(interpolation_max, value_at_corner);
+
+                      }
+                    if ((interpolation_max - cell_average_values[property_index]) > std::numeric_limits<double>::epsilon() &&
+                        (cell_average_values[property_index] - interpolation_min) > std::numeric_limits<double>::epsilon())
+                      {
+                        const double alpha = std::max(std::min((cell_average_values[property_index] - property_minimums[property_index])/(cell_average_values[property_index] - interpolation_min),
+                                                               (property_maximums[property_index]-cell_average_values[property_index])/(interpolation_max - cell_average_values[property_index])), 0.0);
+                        // If alpha >= 1, then using it would make the function grow from the cell average to meet the bounds.
+                        if (alpha < 1.0)
+                          {
+                            c[property_index] *= alpha;
+                            c[property_index][0] += (1-alpha) * cell_average_values[property_index];
+                          }
+                      }
+                  }
               }
           }
         unsigned int index_positions = 0;
+        const auto &mapping = this->get_mapping();
```

Comment:
I do not think there is much saved with this renaming. Just use `this->get_mapping()` below.